client/src/main/java/io/confluent/kafka/schemaregistry/avro/AvroSchema.java

```diff
@@ -576,12 +577,20 @@ private Object toTransformedMessage(
           try {
             Set<String> ruleTags = ctx.rule().getTags();
             if (ruleTags.isEmpty()) {
-              return transform.transform(ctx, fieldCtx, message);
+              Object result = transform.transform(ctx, fieldCtx, message);
+              if (result instanceof byte[]) {
+                result = ByteBuffer.wrap((byte[]) result);
+              }
+              return result;
             } else {
               Set<String> intersect = new HashSet<>(fieldCtx.getTags());
               intersect.retainAll(ruleTags);
               if (!intersect.isEmpty()) {
-                return transform.transform(ctx, fieldCtx, message);
+                Object result = transform.transform(ctx, fieldCtx, message);
+                if (result instanceof byte[]) {
+                  result = ByteBuffer.wrap((byte[]) result);
+                }
+                return result;
               }
```

Comment:
Just wondering, line 558, if intersect.isEmpty() is true, do we need to check 
if (result instanceof byte[]) {
      result = ByteBuffer.wrap((byte[]) result);
}
in else.

Reply:
No, we'll just fall throw to the `return message` at the end.  Thanks for asking?